opte/src/lib.rs

```diff
@@ -242,12 +242,8 @@ impl LogProvider for KernelLog {
             LogLevel::Error => ddi::CE_WARN,
         };
 
-        unsafe {
-            ddi::cmn_err(
-                cmn_level,
-                cstr_core::CString::new(msg.to_string()).unwrap().as_ptr(),
-            )
-        }
+        let msg_arg = alloc::ffi::CString::new(msg.to_string()).unwrap();
```

Comment:
The extra `.to_string()` conversion here seems unnecessary, I think. `CString::new` takes anything convertible to a `Vec<u8>`, which includes `&str`.

Reply:
Done.